packages/waku/src/router/client.ts

```diff
@@ -31,13 +31,37 @@ import {
   LOCATION_ID,
 } from './common.js';
 import type { RouteProps, ShouldSkip } from './common.js';
+import type { createPages } from './create-pages.js';
+import type { CollectPaths } from './inferred-router-types.js';
```

Comment:
I would like to make this file depends on create-pages.
This file is paired with `define-router.ts` which has no notions about "pages".

Reply:
I understand...

I am working in this file because I need to override the type of `to` for the Link component.

Would you like to me to add an export for "./router" and surface the types [there](https://github.com/dai-shi/waku/blob/main/packages/waku/package.json#L18-L53)? 

This client.ts file would then import the `InferredPath` (or whatever it gets renamed to) from that router file.

Reply:
> I need to override the type of `to` for the Link component.

That's fine. A change in this file is acceptable. Just make it agnostic to pages.

Reply:
I'm thinking something like:
```ts
export interface RoutePaths {
  // to be overridden by users
}
```
Or its holder.

Reply:
Oh I get it! So then would you want to export the generic to pull them from the router and give an example like this?
```ts
import type { PathsForRouter } from 'waku/router';

declare module 'waku/router' {
  interface RouteConfig {
    routes: PathsForRouter<typeof myRouter> | '/some-other-route'; // could also be whatever else they want
  }
}
```

Reply:
Exactly.

Reply:
Ok, updated! Let me know what you think.

I'll update the docs to reflect the changes in the meantime 👍